### DISCUSSION TITLE
Dependency of Resources

### DISCUSSION THREAD

**Discussion by vwake7** [stackql/stackql#398] [Q&A] [answered]
@general-kroll-4-life, What is the best way to handle dependencies in pystackql.  Suppose I want to create a VM in Azure, then I need to create a Resource Group, public ip, NIC before I create the VM. Is there a way to roll back if one of the dependencies fail. Can you please provide the suggested approach.

**Comment by jeffreyaven** [comment, marked as answer]
Hey @vwake7 have a look at this [introducing-model-driven-iac-with-stackql-deploy](https://stackql.io/blog/introducing-model-driven-iac-with-stackql-deploy), new deployment framework using `stackql`, does exactly what you are looking for, let me know if we can help you set something up
